lib/ansible/cli/config.py

```diff
@@ -0,0 +1,179 @@
+        elif self.action == "edit":
+            self.parser.set_usage("usage: %prog edit [options] [-c ansible.cfg]")
+        elif self.action == "update":
+            self.parser.add_option('-s', '--setting', dest='setting', help="config setting, the section defaults to 'defaults'")
+            self.parser.set_usage("usage: %prog update [options] [-c ansible.cfg] -s '[section.]setting=value'")
+
+        self.options, self.args = self.parser.parse_args()
+        display.verbosity = self.options.verbosity
+
+    def run(self):
+
+        super(ConfigCLI, self).run()
+
+        if self.options.config_file:
+            self.config_file = unfrackpath(self.options.config_file, follow=False)
+            self.config = ConfigManager(self.config_file)
+        else:
+            self.config = ConfigManager()
+            self.config_file = self.config.data.get_setting('ANSIBLE_CONFIG')
+            try:
+                if not os.path.exists(self.config_file):
+                    raise AnsibleOptionsError("%s does not exist or is not accessible" % (self.config_file))
+                elif not os.path.isfile(self.config_file):
+                    raise AnsibleOptionsError("%s is not a valid file" % (self.config_file))
+
+                os.environ['ANSIBLE_CONFIG'] = self.config_file
+            except:
+                if self.action in ['view']:
+                    raise
+                elif self.action in ['edit', 'update']:
+                    display.warning("File does not exist, used empty file: %s" % self.config_file)
+
+        self.execute()
+
+    def execute_update(self):
+        '''
+        Updates a single setting in the specified ansible.cfg
+        '''
+        raise AnsibleError("Option not implemented yet")
+
+        if self.options.setting is None:
+            raise AnsibleOptionsError("update option requries a setting to update")
+
+        (entry, value) = self.options.setting.split('=')
+        if '.' in entry:
+            (section, option) = entry.split('.')
+        else:
+            section = 'defaults'
+            option = entry
+        subprocess.call([
+            'ansible',
+            '-m','ini_file',
+            'localhost',
+            '-c','local',
+            '-a','"dest=%s section=%s option=%s value=%s backup=yes"' % (self.config_file, section, option, value)
+        ])
+
+    def execute_view(self):
+        '''
+        Displays the current config file
+        '''
+        try:
+            self.pager(open(self.config_file).read())
+        except Exception as e:
+            raise AnsibleError("Failed to open config file: %s" % str(e))
+
+    def execute_edit(self):
+        '''
+        Opens ansible.cfg in the default EDITOR
+        '''
+        raise AnsibleError("Option not implemented yet")
+        try:
+            editor = shlex.split(os.environ.get('EDITOR','vi'))
+            editor.append(self.config_file)
+            subprocess.call(editor)
```

Comment:
Note, anytime we use shlex.split, we have to do a song and dance of converting to byte strings on python2 and text strings on python3. (The reason is that py3 only handles text string input and python2.6 has a bug when it gets text strings as input).  For perfect resilience we should give subprocess.call() a byte string as well (probably doesn't matter as commands probably don't exist in paths with non-ascii parts).

something like:
``` python
        try:
            editor_command = to_native(os.environ.get('EDITOR', 'vi'))
            b_editor = shlex.split(editor_command)
            if PY3:
                b_editor = [to_bytes(elem, errors='surrogate_or_strict') for elem in b_editor] 
            b_editor.append(to_bytes(self.config_file, errors='surrogate_or_strict'))
            subprocess.call(b_editor)